website/content/en/preview/contributing/development-guide.md

```diff
@@ -21,42 +21,64 @@ The following tools are required for contributing to the Karpenter project.
 
 ### Setup / Teardown
 
-Based on how you are running your Kubernetes cluster, follow the [Environment specific setup](#environment-specific-setup) to configure your environment before you continue. You can choose to either run the Karpenter controller locally on your machine, pointing to the Kubernetes cluster specified in your `~/.kube/config` or inside the Kubernetes cluster specified in your `~/.kube/config` deployed with [Helm](https://helm.sh/).
+#### Container Repository
 
-#### Locally
+For local development on Karpenter you will need a Docker repo which can manage your images for Karpenter components.
+You can use the following command to provision an ECR repository. We recommend using a single "dev" repository for
+development across multiple projects, and to use specific image hashes instead of image tags.
+
+```bash
+aws ecr create-repository \
+    --repository-name dev \
+    --image-scanning-configuration scanOnPush=true \
+    --region "${AWS_DEFAULT_REGION}"
+```
 
-Once you have your environment set up, run the following commands to run the Karpenter Go binary against the Kubernetes cluster specified in your `~/.kube/config`
+Once you have your ECR repository provisioned, configure your Docker daemon to authenticate with your newly created repository.
 
 ```bash
-make run
+export KO_DOCKER_REPO="${AWS_ACCOUNT_ID}.dkr.ecr.${AWS_DEFAULT_REGION}.amazonaws.com/dev"
+aws ecr get-login-password --region "${AWS_DEFAULT_REGION}" | docker login --username AWS --password-stdin "${KO_DOCKER_REPO}"
 ```
 
-#### Inside a Kubernetes Cluster
+#### Kubernetes Cluster
+
+You can refer to the [Getting Started Guide]({{<ref "../getting-started/getting-started-with-karpenter#create-a-cluster" >}}) for details on how to setup a cluster and IAM permissions to deploy Karpenter.
+
+#### Test Dependencies
 
-Once you have your environment set up, to install Karpenter in the Kubernetes cluster specified in your `~/.kube/config`  run the following commands.
+Karpenter requires [Kubebuilder's `envtest` dependencies](https://book.kubebuilder.io/reference/envtest.html) to run functional testing. Run the following command to install the test dependencies.
 
 ```bash
-make apply # Install Karpenter
-make delete # Uninstall Karpenter
+make toolchain # install test dependencies
 ```
 
-### Developer Loop
+#### Environment Variables
```

Comment:
Consider moving this section up, since you refer to some of these variables in previous steps.